packages/compartment-mapper/src/policy.js

```diff
@@ -115,60 +120,63 @@ export const detectAttenuators = policy => {
+export const dependencyAllowedByPolicy = (canonicalName, packagePolicy) => {
   return !!policyLookupHelper(packagePolicy, 'packages', canonicalName);
 };
 
 /**
- * Returns the policy applicable to the canonicalName of the package
- *
+ * Generates the {@link SomePackagePolicy} value to be used in {@link CompartmentDescriptor.policy}
  * @overload
- * @param {PackageNamingKit} namingKit - a key in the policy resources spec is derived from these
- * @param {SomePolicy} policy - user supplied policy
- * @returns {SomePackagePolicy} packagePolicy if policy was specified
+ * @param {CanonicalName | typeof ATTENUATORS_COMPARTMENT} label
+ * @param {object} options Options
+ * @param {SomePolicy} options.policy User-supplied policy
+ * @param {SomePackagePolicy} [options.packagePolicy] Package policy, if already known
+ * @returns {SomePackagePolicy} Package policy from `policy` or empty object; returns `params.packagePolicy` if provided
  */
-
 /**
- * Returns `undefined`
+ * Generates the {@link SomePackagePolicy} value to be used in {@link CompartmentDescriptor.policy}
  *
  * @overload
- * @param {PackageNamingKit} namingKit - a key in the policy resources spec is derived from these
- * @param {SomePolicy} [policy] - user supplied policy
- * @returns {SomePackagePolicy|undefined} packagePolicy if policy was specified
+ * @param {CanonicalName | typeof ATTENUATORS_COMPARTMENT} label
+ * @param {object} [options] Options
+ * @param {SomePolicy} [options.policy] User-supplied policy
+ * @param {SomePackagePolicy} [options.packagePolicy] Package policy, if already known
+ * @returns {SomePackagePolicy|undefined} Package policy from `policy` or empty object; returns `params.packagePolicy` if provided
  */
-
 /**
- * Returns the policy applicable to the canonicalName of the package
+ * Generates the {@link SomePackagePolicy} value to be used in {@link CompartmentDescriptor.policy}
  *
- * @param {PackageNamingKit} namingKit - a key in the policy resources spec is derived from these
- * @param {SomePolicy} [policy] - user supplied policy
+ * @param {CanonicalName | typeof ATTENUATORS_COMPARTMENT} label
+ * @param {object} [options] Options
+ * @param {SomePolicy} [options.policy] User-supplied policy
+ * @param {SomePackagePolicy} [options.packagePolicy] Package policy, if already known
+ * @returns {SomePackagePolicy|undefined} Package policy from `policy` or empty object; returns `params.packagePolicy` if provided
  */
-export const getPolicyForPackage = (namingKit, policy) => {
-  if (!policy) {
-    return undefined;
-  }
-  if (namingKit.isEntry) {
-    return policy.entry;
-  }
-  const canonicalName = generateCanonicalName(namingKit);
-  if (canonicalName === ATTENUATORS_COMPARTMENT) {
-    return { defaultAttenuator: policy.defaultAttenuator, packages: 'any' };
+export const makePackagePolicy = (
+  label,
+  { policy, packagePolicy = undefined } = {},
+) => {
+  if (packagePolicy !== undefined) {
```

Comment:
how would a package policy be already known but still go through this function? What is the point of doing it here instead of checking at call location?